modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentIdFieldMapper.java

```diff
@@ -140,13 +146,28 @@ public String getParentName() {
         return parentName;
     }
 
+    public Query getParentFilter() {
+        return new TermQuery(new Term(name().substring(0, name().indexOf('#')), parentName));
+    }
     /**
      * Returns the children names associated with this mapper.
      */
     public Collection<String> getChildren() {
         return children;
     }
 
+    public Query getChildFilter(String type) {
+        return new TermQuery(new Term(name().substring(0, name().indexOf('#')), type));
+    }
+
+    public Query getChildrenFilter() {
+        BooleanQuery.Builder builder = new BooleanQuery.Builder();
+        for (String child : children) {
+            builder.add(getChildFilter(child), BooleanClause.Occur.SHOULD);
```

Comment:
That doesn't work because we would be AND-ing the child filter together and a doc just need to match with one of the parent's child types. (also see the old logic, which SHOULD-ing everything together)